overlord/overlord.go

```diff
@@ -68,6 +68,10 @@ var (
 	configstateInit = configstate.Init
 )
 
+var pruneTickerC = func(t *time.Ticker) <-chan time.Time {
```

Comment:
this probably merits a comment:

// hook for testing prune logic

can be added in a follow up if this turns green at first try